integrationtests/gitjob/controller/controller_test.go

```diff
@@ -616,6 +652,39 @@ var _ = Describe("GitJob controller", func() {
 				}, "30s", "1s").Should(Equal(stableCommit))
 			})
 		})
+
+		When("WebhookCommit changes and user clicks ForceUpdateGeneration", func() {
+			BeforeEach(func() {
+				gitRepoName = "disable-polling-commit-change-force-update"
+				webhookCommit = "af6116a6c5c3196043b4a456316ae257dad9b5db"
+				expectedCommit = stableCommit
+				// user clicks ForceUpdate 2 times
+				// This exercises possible race conditions in the reconciler
+				forceUpdateGeneration = 2
+			})
+
+			It("creates a new Job", func() {
+				Eventually(func() error {
+					jobName := names.SafeConcatName(gitRepoName, names.Hex(repo+webhookCommit, 5))
+					return k8sClient.Get(ctx, types.NamespacedName{Name: jobName, Namespace: gitRepoNamespace}, &job)
```

Comment:
If more than one job were created, how would we detect it?

Reply:
The job name depends on the repopath+commit, so unless the commit changes...we'll have the same job all the time. 
We can see from time to time `Already Exist` errors when trying to re-create the job under race condition circumstances.